app/code/core/Mage/SalesRule/Model/Quote/Discount.php

```diff
@@ -134,7 +134,7 @@ protected function _aggregateItemDiscount($item)
     /**
      * Add discount total information to address
      *
-     * @return $this
+     * @return $this|array
```

Comment:
I cannot see how it can return an array.

Reply:
To match `parrent::fetch(()`, that returns an array.